lib/puppet/provider/vcsrepo/git.rb

```diff
@@ -407,6 +422,33 @@ def init_repository
     end
   end
 
+  # @!visibility private
+  def configure_sparse_checkout
+    raise("Cannot set includes on a #{@resource.value(:ensure)} repository") if ensure_bare_or_mirror? || bare_exists?
+
+    git_ver = git_version
+    if Gem::Version.new(git_ver) >= Gem::Version.new('2.25.0')
+      # sparse-checkout command was introduced in version 2.25.0.
+      at_path do
+        args = ['sparse-checkout', 'set', '--no-cone'] + @resource.value(:includes)
+        exec_git(*args)
+      end
+    else
+      at_path do
+        exec_git('config', '--local', '--bool', 'core.sparseCheckout', 'true')
+
+        # Includes may be an Array or a String
+        File.open(File.join(@resource.value(:path), '.git/info/sparse-checkout'), 'w') do |f|
```

Comment:
Same here, a relative path is fine:

```suggestion
        File.open('.git/info/sparse-checkout', 'w') do |f|
```